Short-Answer/Algorithms_Answers.md

```diff
@@ -2,14 +2,16 @@
 
 ## Exercise I
 
-a)
+a) O(n) - loop will take longer to finish as n increases
 
 
-b)
+b) O(n^2) - there is a loop within a loop
```

Comment:
Nice job with these.

For b), while there is a loop within a loop you have to consider what happens to `j` on each iteration of the inner while-loop:

```
b)  sum = 0
    for i in range(n):
      j = 1
      while j < n:
        j *= 2
        sum += 1
```

`j` doubles in size on each iteration for `j < n`, therefore we half the number of elements we traverse through on each iteration of the inner while-loop. This would produce a runtime complexity of `O(log n)`

If you account for the outer for-loop, you get `O(n * log n)`